libs/partners/openai/langchain_openai/embeddings/base.py

```diff
@@ -518,24 +485,19 @@ def embed_documents(
             List of embeddings, one for each text.
         """
         if not self.check_embedding_ctx_length:
-            embeddings: List[List[float]] = []
-            for text in texts:
-                response = self.client.create(
-                    input=text,
-                    **self._invocation_params,
-                )
-                if not isinstance(response, dict):
-                    response = response.dict()
-                embeddings.extend(r["embedding"] for r in response["data"])
-            return embeddings
+            return self._get_embeddings(texts, chunk_size=chunk_size)
 
-        # NOTE: to keep things simple, we assume the list may contain texts longer
-        #       than the maximum context and use length-safe embedding function.
-        engine = cast(str, self.deployment)
-        return self._get_len_safe_embeddings(texts, engine=engine)
+        num_texts = len(texts)
+        split_tokens, embedding_to_text_indices = self._tokenize_and_split(texts)
+        if len(split_tokens) == num_texts:
```

Comment:
Can we add a test where one of the texts is `""`, and one of the texts is worth 2 splits? I think this will do funky things.